core/txpool/blobpool/blobpool.go

```diff
@@ -871,6 +885,70 @@ func (p *BlobPool) Reset(oldHead, newHead *types.Header) {
 	basefeeGauge.Update(int64(basefee.Uint64()))
 	blobfeeGauge.Update(int64(blobfee.Uint64()))
 	p.updateStorageMetrics()
+
+	// Perform the conversion logic at the fork boundary
+	if !p.chain.Config().IsOsaka(oldHead.Number, oldHead.Time) && p.chain.Config().IsOsaka(newHead.Number, newHead.Time) {
+		// Evicts all transactions from the pool. The pool should, at this stage,
+		// contain only legacy blob transactions. Evicting any new-format blob
+		// transactions is also safe, as they will be re-injected later.
+		txs := make(map[common.Address]map[uint64]uint64)
+		for p.stored > 0 {
+			sender, nonce, id := p.drop(false)
+			if txs[sender] == nil {
+				txs[sender] = make(map[uint64]uint64)
+			}
+			txs[sender][nonce] = id
+		}
+		// Initiate the background conversion thread, the mutex is not required
+		// and won't block any pool operation.
+		go func() {
+			var (
+				start   = time.Now()
+				success int
+				fail    int
+			)
+			for addr, list := range txs {
```

Comment:
We could randomize the order in which we traverse addresses. The advantage would be that at the system level, different nodes would convert different blobs, sharing the load, resulting in a faster overall conversion of the pending part of the distributed mempool.

Reply:
Although we use a map, so it is already randomized. Might be enough.

Reply:
Yep, the randomness from map should be sufficient